src/mame/seta/speglsht.cpp

```diff
@@ -376,23 +372,24 @@ void speglsht_state::machine_reset()
 void speglsht_state::video_start()
 {
 	m_bitmap = std::make_unique<bitmap_ind16>(512, 512);
+	save_item(NAME(m_videoreg));
 }
 
-#define PLOT_PIXEL_RGB(x,y,r,g,b)   if(y>=0 && x>=0 && x<512 && y<512) \
+#define PLOT_PIXEL_RGB(x, y, r, g, b)   if (y >= 0 && x >= 0 && x < 512 && y < 512) \
 { \
-		bitmap.pix(y, x) = (b) | ((g)<<8) | ((r)<<16); \
+		bitmap.pix(y, x) = (b) | ((g) << 8) | ((r) << 16); \
 }
 
 uint32_t speglsht_state::screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect)
 {
-	int dy=(m_videoreg&0x20)?(256*512):0; //visible frame
+	int const dy = BIT(m_videoreg, 5) ? (256 * 512) : 0; //visible frame
 
-	for(int y=0;y<256;y++)
+	for (int y = 0; y < 256; y++)
 	{
-		for(int x=0;x<512;x++)
+		for (int x = 0; x < 512; x++)
 		{
-			int tmp=dy+y*512+x;
-			PLOT_PIXEL_RGB(x-67,y-5,(m_framebuffer[tmp]>>0)&0xff,(m_framebuffer[tmp]>>8)&0xff,(m_framebuffer[tmp]>>16)&0xff);
+			int const tmp = dy + y * 512 + x;
+			PLOT_PIXEL_RGB(x - 67, y - 5, (m_framebuffer[tmp] >> 0) & 0xff, (m_framebuffer[tmp] >> 8) & 0xff, (m_framebuffer[tmp] >> 16) & 0xff);
 		}
 	}
```

Comment:
This `PLOT_PIXEL_RGB` macro has no right to live: it should respect cliprect when looping, and do pitch start base addressing before inner x instead. At bare minimum unrolling is desired.